### ISSUE TITLE
Add article subtitles to title string in outgoing emails

### ISSUE THREAD

**Issue by jmacgreg** [pkp/pkp-lib#3675]
It looks like submission-related emails only include titles and not subtitles - they should also include subtitles, if included.

**Comment by asmecher** [comment]
@jmacgreg, how do you think we should format these? I think just gluing the title and subtitle together will lead to ugliness. A few examples might help straighten it out.

**Comment by jmacgreg** [comment]
@asmecher I think the way we did this in the old TOC (before @NateWr got into doing more involved theming) would work, which was just to concatenate them together: $title . ": " . $subTitle. No need to get fancy here. The helper text under the subtitle field says: "The optional subtitle will appear after a colon (:), following the main title."

**Comment by asmecher** [comment]
Hmm, just thinking about languages where `${title}: ${subtitle}` might not be appropriate. Maybe we need a locale key for title concatenation? Thinking:

    <message key="article.title.concatenated">{$title}: {$subtitle}</message>

We'd need to use that *only* when both title and subtitle are set, so as not to leave an [ehrm] dangling colon.

**Comment by NateWr** [comment]
I like the idea of a locale string. It does raise some issues with the logic around when to use a `:` and when not to.

We have `Submission::getLocalizedFullTitle()` which does this work: https://github.com/pkp/pkp-lib/blob/master/classes/submission/Submission.inc.php#L464-L472

And it relies on `PKPString::concatTitleFields()` which does some English-language-specific logic for determining when a `:` is appropriate: https://github.com/pkp/pkp-lib/blob/master/classes/core/PKPString.inc.php#L746-L768

So, we could have three locale keys:

```
<message key="article.title.concatenated">{$title}: {$subtitle}</message>
<message key="article.title.concatenatedWithoutSeparator">{$title} {$subtitle}</message>
<message key="article.title.concatenationConditions">?!/&</message>
```

But I fear getting into even more locale-by-locale particularities. Maybe a language needs more than two separator cases, etc.

One possibility would be to just hook the `PKPString::concatTitleFields()` so that languages with their own rules could override it as needed:

```
Hook::register('Submission::ConcatenateTitle', array(&$fullTitle, $submission));
return $fullTitle;
```

**Comment by fgnievinski** [comment]
related issue: #4021

**Comment by fgnievinski** [comment]
concatTitleFields is already used in getLocalizedFullTitle:

https://github.com/pkp/pkp-lib/blob/4e3ffe04cc7991294b26d4a76279a16976b4951d/classes/publication/PKPPublication.inc.php#L63

which is already being used to push metadata with full title, as per #5066.  

so it seems the same could be done here?

**Comment by NateWr** [comment]
I think `getLocalizedFullTitle()` should be used in all cases except where metadata explicitly asks for the title/subtitle to be passed separately. If you spot somewhere it's not being used but should, it could probably be changed.

**Comment by NateWr** [comment]
This is being fixed in #5730.
